pkg/cli/upgrade/upgrade.go

```diff
@@ -102,7 +102,7 @@ func NewUpgrade(cfg *Config, l *zap.SugaredLogger) (*Upgrade, error) {
 
 	var kubeClient kubernetes.KubernetesConnector
 	if cfg.InCluster {
-		k, err := kubernetes.NewInCluster(cli.l)
+		k, err := kubernetes.NewInCluster(cli.l, context.Background(), nil)
```

Comment:
this is an issue - because when user terminates the process via `Esc` or `Ctrl+C` the CLI context is being canceled as well, but cache will continue running.

Reply:
there is no cache if no cacheOptions are specified

Reply:
to make it less confusing, now passing nil instead of an empty context. wdyt?